R/align_lineage.R

```diff
@@ -0,0 +1,100 @@
+#' This function aligns all sequences incliding germline within one clonal lineage. After building
+#' clonal lineage and germline, the next step isto analyze the degree of mutation and maturity
+#' of each clonal lineage. This allows you to find high mature cells and cells with a large number of offspring.
+#' The phylogenetic analysis will find mutations that increase the affinity of BCR.
+#' Making alignment of the sequence is the first step towards sequence analysis including BCR.
+#'
+#' @concept align_lineage
+#'
+#' @aliases repAlignLineage
+#'
+#' @importFrom magrittr %>% %<>%
+#' @importFrom stringr str_extract_all
+#' @importFrom ape as.DNAbin muscle
+#' @importFrom parallel mclapply detectCores
+
+#' @description Aligns all sequences incliding germline within one clonal lineage
+#'
+#' @usage
+#'
+#' repAlignLineage(.data)
+#'
+#' @param .data The data to be processed. Can be \link{data.frame}, \link{data.table}
+#' or a list of these objects.
+#'
+#' It must have columns in the immunarch compatible format \link{immunarch_data_format}, and also
+#' must contain Sequence.germline column, which is added by repGermline() function.
+#'
+#' @return
+#'
+#' DNAbin object with alignment if there was single input dataframe,
+#' or list of DNAbin objects if the input was a list of dataframes
+#'
+#' @examples
+#'
+#' data(immdata)
+#'
+#' immdata$data %>%
+#'   top(2000) %>% # reduce the dataset to save time on examples
+#'   repGermline() %>%
+#'   repAlignLineage()
+#' @export repAlignLineage
+repAlignLineage <- function(.data) {
+  require_system_package("muscle", error_message = paste0(
+    "repAlignLineage requires MUSCLE app to be installed!\n",
+    "Please download it from here: https://github.com/rcedgar/muscle/releases/latest\n",
+    "or install it with your system package manager (such as apt or dnf)."
+  ))
+  if (inherits(.data, "list")) {
+    .validate_repertoires_data(.data)
+    .data %>%
+      lapply(function(sample_data) {
+        sample_data %>%
+          as_tibble() %>%
+          align_single_df() %>%
+          return()
+      }) %>%
+      return()
+  } else {
+    .data %>%
+      as_tibble() %>%
+      align_single_df() %>%
+      return()
```

Comment:
Please avoid using right-side hand assignment, especially for returns

Reply:
Without pipes it will be a long chain of nested brackets; I think, sequence of `%>%` makes it more readable, and style guides allow it

Reply:
I didn't mean that you can't use the `%>%` operator at all. Just don't use it to pass data to return
```suggestion
      .data %<>%
      lapply(function(sample_data) {
        sample_data %>%
          as_tibble() %>%
          align_single_df()
      })
      return(.data)
```